[Xzh119] iP

Xzh119 opened the pull request:
### Samantha

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Samantha, named after the emotional AI in the movie HER, frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2425S2/ip/pull/450).
2. double-click it.
3.  add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
add Level 0

Commit:
Add Echo

Commit:
Add list and add ability

Commit:
Add a Task class and Mark as Done feature

Commit:
Add Todo, Event and Deadline class

Commit:
Add Automated Text UI Testing

Commit:
Add exception to handle errors

Commit:
Add Delete feature

blobfish465 commented on src/main/java/Samantha.java:
```diff
@@ -0,0 +1,140 @@
+            } else {
+                if (taskSum > 99) {
+                    System.out.println("    List is full.");
+                }else {
```
Should there be a space between { and else?

blobfish465 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
```
Good use of a camelCase for method naming.

blobfish465 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Good use of a boolean naming convention.

blobfish465 commented on src/main/java/SamanthaException.java:
```diff
@@ -0,0 +1,5 @@
+public class SamanthaException extends Exception {
+    public SamanthaException(String message) {
+        super(message);
+    }
+}
```
Should there be more specific exceptions e.g. NumberFormatException, IndexOutOfBoundsException ?

blobfish465 commented on src/main/java/Samantha.java:
```diff
@@ -0,0 +1,140 @@
+import java.util.Scanner;
+import java.util.ArrayList;
```
You might want to leave a line below this statement before defining your public class

blobfish465 reviewed (commented):
Overall I found your code easy to read for the most part, except for some areas where there were minor coding standard errors such as missing spacing and lines. You might want to consider splitting some single lines into several lines so it is easier to read.

Commit:
Add the ability to save the tasks in the hard disk automatically

Commit:
Try merge test

Commit:
Merge branch 'master' into branch-Level-7

# Conflicts:
#	src/main/java/Task.java

Commit:
Fix merge problem

Commit:
Add dates and times

Commit:
Test a

Commit:
Test b

Commit:
Test c

Commit:
Merge branch 'master' of https://github.com/Xzh119/ip

* 'master' of https://github.com/Xzh119/ip:
  Test b
  Test a

Taoseeker commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,48 @@
+        String type = parts[0];
+
+        switch (type) {
+            case "T":
```
Maybe the indentation for `case` clauses could be removed according to coding standard?

Taoseeker commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,48 @@
+public class Task {
```
Should `Task` be an abstract class instead?

Taoseeker commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+        String[] parts = line.split(" \\| ");
+        boolean isDone = parts[1].equals("1");
+        Deadline task = new Deadline(parts[2], parts[3]);
+        if (isDone) task.markAsDone();
```
Maybe `if` should be followed by a block?

Taoseeker reviewed (commented):
The code LGTM, despite some possible improvement in following coding standard. Additaionally, I like how the naming explain the usage of a field/method clearly.

Commit:
Make the code more OOP

Commit:
Divide classes into packages

Commit:
Merge branch test

Commit:
Merge test 2

Commit:
Merge test 3

Commit:
Merge test 4

Commit:
Merge branch 'branch-Test-C'

Commit:
Add comment for Parser class

Commit:
Merge branch 'add-gradle-support'

Commit:
Set up gradle

Commit:
Add test for Ui and Parser

Commit:
Package a jar file and add a new release

Commit:
Add JavaDoc comments

Commit:
Tweak the code to comply with a coding standard

Commit:
Add the ability to find tasks

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branches and fix conflicts

Commit:
Add a GUI to the chatbot

Commit:
Merge branch 'branch-Level-10'

Commit:
Fix potential conflicts

Commit:
Change jar name

Commit:
Add assertions to DialogBox and Main

Commit:
Add assertions to Parser for input validation

The `parseCommand` method does not have assertions to verify
the validity of inputs. This may lead to unexpected runtime errors.

Add assertions to ensure inputs meet expected conditions. Assertions
check for null command and properly formatted commands.

This helps catch potential bugs early and improves code robustness by
preventing invalid operations before they can cause failures.

Commit:
Improve Code Quality

Xzh119 committed:
Merge pull request #2 from Xzh119/branch-A-Assertions

Merge Branch-a-assertions

Commit:
Merge branch 'master' into branch-A-CodeQuality

Xzh119 committed:
Merge pull request #3 from Xzh119/branch-A-CodeQuality

Improve Code Quality

Xzh119 committed:
Revert "Improve Code Quality"

Xzh119 committed:
Merge pull request #4 from Xzh119/revert-3-branch-A-CodeQuality

Revert "Improve Code Quality"

Xzh119 committed:
Merge pull request #5 from Xzh119/branch-A-CodeQuality

Merge branch 'master' into branch-A-CodeQuality

Commit:
Friendlier syntax for commands

Commit:
Merge branch 'branch-C-FriendlierSyntax'

Commit:
Fix the bug of a redundant line generated by list command

It needs to be change as it affects the purpose of listing.

It is fixed by preventing the "\n" at the end.

It is done this way because it is reasonable and easy.

Commit:
Add Ui picture

Commit:
Add a User Guide

Commit:
Add comments for Task

Commit:
Add a new exception
